waimak-core/src/main/scala/com/coxautodata/waimak/dataflow/DFExecutorPriorityStrategies.scala

```diff
@@ -0,0 +1,99 @@
+package com.coxautodata.waimak.dataflow
+
+/**
+  * Defines various priority strategies for [[ DataFlowExecutor ]] to use.
+  *
+  * Created by Alexei Perelighin on 24/08/2018.
+  */
+object DFExecutorPriorityStrategies {
+
+  type actionQueue[C] = Seq[DataFlowAction[C]]
+
+  type priorityStrategy[C] = actionQueue[C] => actionQueue[C]
+
+  /**
+    * Default strategy, at the moment it will not do anything.
+    *
+    * @tparam C
+    * @return
+    */
+  def defaultPriorityStrategy[C]: PartialFunction[actionQueue[C], actionQueue[C]] = doNothing[C]
+
+  /**
+    * Preserves the order of the actions in which they are defined, but at first will give preference to loaders. If
+    * there are no loaders, keeps the order.
+    *
+    * @tparam C
+    * @return
+    */
+  def preferLoaders[C]: PartialFunction[actionQueue[C], actionQueue[C]] = takeLoaders[C] orElse doNothing[C]
+
+  /**
+    * With Spark, waimak writers would usually force execution of the DAG and will produce outputs, while other waimak actions
+    * could be preparing the steps of the DAG. In order to load the Spark Executors with work this strategy:
+    *  1) will first choose only writers, as those are most likely to execute the DAG
+    *  2) if there were no writers, it will choose only actions with inputs, as those will to DAG execution faster
+    *  3) if there were no writers and actions with inputs leaves them as they are
+    * @tparam C
+    * @return
+    */
+  def fastTrackToDAG[C]: PartialFunction[actionQueue[C], actionQueue[C]] = takeWriters[C] orElse takeWithInputs[C] orElse doNothing[C]
```

Comment:
the name fastTrackToDAG isn't very clear about what this actually does. Really what it's doing is trying to prioritise writers, so maybe a name which suggests that, or something like depthFirst (as it's doing a depth-first traversal of the Waimak DAG). Thoughts?

Reply:
renamed into 'raceToOutputs'